### DISCUSSION TITLE
How do I delete a node so that all the other nodes "move up"? (picture of what I am talking about in the description)

### DISCUSSION THREAD

**Discussion by adamfreeplaneuser01** [freeplane/freeplane#1472] [Help] [answered]
Hey. So I am trying to delete that node without everything else being deleted. I'm sure there's a simple way to delete only that node but I can't figure it out. Thanks for the help!

<img width="669" alt="ijijdsfdsfa" src="https://github.com/freeplane/freeplane/assets/147225942/74332fa4-678e-450a-8288-a8eba45bd521">

**Comment by sifran-github** [comment]
Whenever I want to do that, I do the following steps: 

1. Select all the child nodes.
2. Move them one level up by : edit -> move and sort -> move nodes to the left. I use a keyboard shortcut for that, so its quick.
3. Delete the remaining parent node.

**Comment by macmarrum** [comment]
I use [extractAndDeleteNode.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/scripts/extractAndDeleteNode.groovy) for the task, assigned to `Shift+Del`.

**Reply by adamfreeplaneuser01** [reply]
Hi, so I cannot get that script to show up in the script tab. I'm able to get other scripts to show up though so I'm confused
<img width="569" alt="2342423" src="https://github.com/freeplane/freeplane/assets/147225942/5613a32f-7f5b-4eff-b04a-7512cfc80316">
<img width="299" alt="ewkrjrewjrrwe" src="https://github.com/freeplane/freeplane/assets/147225942/8647b389-96e8-4976-a717-705ded04a68f">

**Reply by sifran-github** [reply, marked as answer]
The script is defined to be in the menu: "/menu_bar/Mac1"

if you want the script to be in the Scripts menu, change the first line from:
// @ExecutionModes({ON_SINGLE_NODE="/menu_bar/Mac1"})

to:
// @ExecutionModes({ON_SINGLE_NODE})

**Reply by adamfreeplaneuser01** [reply]
Thanks, this works perfectly!
